src/index.js

```diff
@@ -0,0 +1,181 @@
+  } else if (currentTemp >= 59) {
+    landscape.textContent = '🌾🌾_🍃_🪨__🛤_🌾🌾🌾_🍃';
+  } else {
+    landscape.textContent = '🌲🌲⛄️🌲⛄️🍂🌲🍁🌲🌲⛄️🍂🌲';
+  }
+}
+
+// WAVE 3
+const updateCityName = () => {
+  cityNameInput.addEventListener("input", updateValue);
+
+  function updateValue(e) {
+    headerCityName.textContent = e.target.value;
+  }
+}
+
+// WAVE 4
+const findLatitudeAndLongitude = async (query) => {
+  try {
+    const response = await axios
+      .get('http://127.0.0.1:5000/location', {
+        params: {
+          q: query,
+          format: 'json'
+        }
+      });
+
+    const latitude = response.data[0].lat;
+    const longitude = response.data[0].lon;
+    console.log('success in findLatitudeAndLongitude!', 'latitude:', latitude, 'longitude:', longitude);
+
+    return await findWeather(latitude, longitude);
+  } catch (error) {
+    console.log(error);
+    return null;
+  }
+}
+
+const findWeather = async (lat, lon) => {
+  try {
+    const response = await axios
+      .get('http://127.0.0.1:5000/weather', {
+        params: {
+          format: 'json',
+          lat: lat,
+          lon: lon,
+        }
+      });
+
+    console.log('success in findWeather!', response.data);
+    return response.data.main.temp;
+  } catch (error) {
+    console.log('error in findWeather!');
+    return null;
+  }
+}
+
+// WAVE 5
+const changeSky = () => {
+  skySelect.addEventListener('change', () => {
+    gardenSection.classList.remove('sunny','cloudy','rainy','snowy');
+
+    if (skySelect.value == 'default') {
+      sky.textContent = '';
+    } else if (skySelect.value == 'sunny') {
+      sky.textContent = `☁️ ☁️ ☁️ ☀️ ☁️ ☁️`;
+      gardenSection.classList.add('sunny');
+    } else if (skySelect.value == 'cloudy') {
+      sky.textContent =`☁️☁️ ☁️ ☁️☁️ ☁️ 🌤 ☁️ ☁️☁️`;
+      gardenSection.classList.add('cloudy');
+    } else if (skySelect.value == 'rainy') {
+      sky.textContent = `🌧🌈⛈🌧🌧💧⛈🌧🌦🌧💧🌧🌧`;
+      gardenSection.classList.add('rainy');
+    } else if (skySelect.value == 'snowy') {
+      sky.textContent = `🌨❄️🌨🌨❄️❄️🌨❄️🌨❄️❄️🌨🌨`;
+        gardenSection.classList.add('snowy');
+    }
+  });
+}
+
+// WAVE 6
+const resetCityName = () => {
+  cityNameReset.addEventListener('click', () => {
+    cityNameInput.value = '';
+    headerCityName.textContent = '';
+  });
+}
+
+
+// Registering event handlers
+const registerEventHandlers = (event) => {
+  increaseTempControl.addEventListener('click', () => {
+    updateTemp(1);
+  });
+
+  decreaseTempControl.addEventListener('click', () => {
+    updateTemp(-1);
+  });
+  
+  currentTempButton.addEventListener('click', async () => {
+    const query = headerCityName.textContent;
+    const temp = await findLatitudeAndLongitude(query);
+    let realTimeTemp = Math.round(temp);
+    
+    if (realTimeTemp !== null) {
+      tempValue.textContent = realTimeTemp; 
+      updateTempColor(realTimeTemp);
+    } else {
+      tempValue.textContent = 'cannot find temperature'; 
+    }
```

Comment:
Is there a reason to keep this an anonymous function or could we move it over to a named helper function to reduce clutter in this function focused on registering the event handlers?